types/bonjour/index.d.ts

```diff
@@ -76,10 +85,10 @@ declare namespace bonjour {
     }
     interface Bonjour {
         (opts?: BonjourOptions): Bonjour;
-        publish(options: ServiceOptions): Service;
+        publish(options: ServiceOptions): LocalService;
```

Comment:
Generally speaking, breaking changes in the types are acceptable (pending some evaluation of the impact) if the old types were incorrect and were going to result in incorrect code.  That seems to be the case for anywhere there used to be a `Service` but there is now a `RemoteService`.

This particular change, on the other hand, seems like it could cause problems.  If someone was assigning the result of `publish` to a variable explicitly typed as `Service`, they see the assignment continue to succeed, but calls to `start`, `stop`, and `published` fail (maybe these things don't happen in practice - I don't know about Bonjour).  This could be mitigated by using the (much uglier) names `BaseService`, `RemoteService` and `Service`, so that the meaning of the name `Service` doesn't change.  From your description, you seem to have prepared yourself for this possibility. :smile:

If you feel very strongly that things must be as they are, I'd appreciate an explanation of why people are unlikely to be broken in real world practice.

Reply:
Makes sense, I've renamed the interfaces as you suggested.